src/chat/UnreadNotice.js

```diff
@@ -54,56 +63,99 @@ export default class UnreadNotice extends React.Component {
+
+      // Notice should go into peeking state
+      clearTimeout(this.hideTimeout);
+      this.hideTimeout = setTimeout(this.hide, HIDE_DELAY);
+    });
+  };
+
   hide = () => {
+    const { unreadCount } = this.props;
     this.state.translateAnimation.setValue(1);
-    Animated.timing(this.state.translateAnimation, hideAnimationConfig).start();
-  };
+    Animated.timing(this.state.translateAnimation,
+      Object.assign({}, hideAnimationConfig, { toValue: unreadCount === 0 ? 0 : PEEKING_FRACTION })
+    ).start(() => {
+      this.setState({
+        noticeState: this.state.translateAnimation._value === 0 ? // eslint-disable-line
+          noticeStates.hidden :
+          noticeStates.peeking
+      });
+    });
+  }
 
-  show = () => {
-    this.state.translateAnimation.setValue(0);
-    Animated.timing(this.state.translateAnimation, showAnimationConfig).start();
+  hidePeekingNotice = () => {
+    this.state.translateAnimation.setValue(PEEKING_FRACTION);
+    Animated.timing(this.state.translateAnimation, hideAnimationConfig).start(() => {
+      this.setState({
+        noticeState: noticeStates.hidden
+      });
+    });
   };
 
   dynamicContainerStyles = () => {
-    const { position, shouldOffsetForInput } = this.props;
-    const translationMultiplier = position === POSITIONS.top ? -1 : 1;
+    const { shouldOffsetForInput } = this.props;
 
     // In narrows where ComposeBox is present translate beyond ComposeBox to avoid blocking it
-    const translateTo = shouldOffsetForInput && position === POSITIONS.bottom ? -40 : 0;
-    const translateFrom = shouldOffsetForInput && position === POSITIONS.bottom ? 0 : 50;
+    const translateFrom = shouldOffsetForInput ? 0 : MAX_TRANSLATION;
+    const translateTo = shouldOffsetForInput ? -MAX_TRANSLATION : 0;
 
     return {
       ...StyleSheet.flatten(styles.unreadContainer),
-      bottom: position === POSITIONS.bottom ? 0 : null,
-      top: position === POSITIONS.top ? 0 : null,
+      bottom: 0,
+      opacity: this.state.translateAnimation.interpolate({
+        inputRange: [0, PEEKING_FRACTION, 1],
+        outputRange: [0, 1, 1]
+      }),
       transform: [
         {
           translateY: this.state.translateAnimation.interpolate({
             inputRange: [0, 1],
-            outputRange: [translationMultiplier * translateFrom, translateTo]
+            outputRange: [translateFrom, translateTo]
           })
         }
       ]
     };
   };
 
   render() {
-    const { count } = this.props;
+    const { unreadCount } = this.props;
 
     return (
       <Animated.View style={this.dynamicContainerStyles()}>
-        <IconDownArrow style={[styles.icon, styles.downArrowIcon]} />
+        <IconDownArrow style={styles.icon} />
         <Text style={styles.unreadText}>
-          {count === 0 ?
-            'No' : count < 100 ?
-              count : '99+'} unread {count === 1 ? 'message' : 'messages'}
+          {unreadCount === 0 ?
+            'No' : unreadCount < 100 ?
```

Comment:
I will do the translation work in a later PR. 
Keep an eye on how I decide to do it, and you'll have to do it for next PRs yourself 👍